hac-core/src/collection/types.rs

```diff
@@ -181,6 +183,39 @@ pub struct Request {
     /// the type of the body to be used, like `application/json` or any other
     /// accepted body type
     pub body_type: Option<BodyType>,
+    /// Examples of what responses may look like, including possible error cases.
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub sample_responses: Vec<SampleResponse>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct SampleResponse {
+    #[serde(default = "temp_sample_resp_name")]
+    pub name: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub body: Option<String>,
+    #[serde(skip_serializing_if = "IndexMap::is_empty")]
+    pub headers: IndexMap<String, Vec<String>>,
```

Comment:
I assume here the idea behind the IndexMap is to preserve order of headers on serialization? I asked this question with the headers for requests, I'm still not sure if this is a huge deal, but I guess its fine